controllers/perconaserverformysql_controller.go

```diff
@@ -243,48 +270,57 @@ func (r *PerconaServerForMySQLReconciler) reconcileReplication(
 
 func reconcileReplicationPrimaryPod(
 	ctx context.Context,
-	cl k8s.APIListPatcher,
+	cl k8s.APIListUpdater,
 	cr *apiv2.PerconaServerForMySQL,
 ) error {
+	l := log.FromContext(ctx).WithName("reconcileReplicationPrimaryPod")
+
 	pods, err := k8s.PodsByLabels(ctx, cl, mysql.MatchLabels(cr))
 	if err != nil {
 		return errors.Wrap(err, "get MySQL pod list")
 	}
+	l.Info(fmt.Sprintf("got %v pods", len(pods)))
 
 	host := orchestrator.APIHost(orchestrator.ServiceName(cr))
 	primary, err := orchestrator.ClusterPrimary(ctx, host, cr.ClusterHint())
 	if err != nil {
-		return errors.Wrap(err, "get cluster from orchestrator")
+		return errors.Wrap(err, "get cluster primary")
 	}
 	primaryAlias := primary.Alias()
+	l.Info(fmt.Sprintf("got cluster primary alias: %v", primaryAlias), "data", primary)
 
 	for i := range pods {
 		pod := &pods[i]
 		if pod.GetLabels()[apiv2.MySQLPrimaryLabel] == "true" {
 			if pod.Name == primaryAlias {
-				// primary is not changed
+				l.Info(fmt.Sprintf("primary %v is not changed. skip", primaryAlias))
 				return nil
 			}
 
-			patch := client.StrategicMergeFrom(pod)
 			k8s.RemoveLabel(pod, apiv2.MySQLPrimaryLabel)
-			if err := cl.Patch(ctx, pod, patch); err != nil {
-				return errors.Wrap(err, "remove label from old primary pod")
+			if err := cl.Update(ctx, pod); err != nil {
+				return errors.Wrapf(err, "remove label from old primary pod: %v/%v",
+					pod.GetNamespace(), pod.GetName())
```

Comment:
maybe Patch is better. why we need Update?

Reply:
ac4c179